src/File.php

```diff
@@ -677,10 +694,26 @@ protected static function decryptData(
         list ($encKey, $authKey) = self::splitKeys($key, $hkdfSalt, $config);
 
         // VERSION 2+ uses BMAC
-        $mac = \sodium_crypto_generichash_init($authKey);
-        \sodium_crypto_generichash_update($mac, $header);
-        \sodium_crypto_generichash_update($mac, $firstNonce);
-        \sodium_crypto_generichash_update($mac, $hkdfSalt);
+        $mac = sodium_crypto_generichash_init($authKey);
+        // Number of pieces that go into MAC (header, first nonce, salt, ciphertext) -> 4
+        if ($config->USE_PAE) {
+            // Number of pieces:
+            sodium_crypto_generichash_update($mac, pack('P', 4));
```

Comment:
This is not 100% compatible, as with `aad` you have 5 pieces. Don't think there is anything one can do here though?